docs/content/rules/sort-modules.mdx

```diff
@@ -0,0 +1,592 @@
+
+```ts
+// Group 1
+interface BasicInformation {
+  firstName: string;
+  lastName: string;
+ }
+
+// Group 2
+interface AgeInformation {
+  age: number;
+  birthDate: Date;
+}
+
+// Group 3
+interface LocationInformation {
+  street: string;
+  city: string;
+}
+
+// Group 4
+function updateAddress(address: string) {}
+function updatePhone(phone?: string) {}
+
+// Group 5
+function editFirstName(firstName: string) {}
+function editLastName(lastName: string) {}
+```
+
+### newlinesBetween
+
+<sub>default: `'ignore'`</sub>
+
+Specifies how new lines should be handled between module member groups.
+
+- `ignore` — Do not report errors related to new lines between object type groups.
+- `always` — Enforce one new line between each group, and forbid new lines inside a group.
+- `never` — No new lines are allowed in object types.
+
+This options is only applicable when `partitionByNewLine` is `false`.
+
+### groups
+
+<sub>
+  type: `Array<string | string[]>`
+</sub>
+<sub>
+  default:
+  ```
+  [
+    'declare-enum',
+    'export-enum',
+    'enum',
+    ['declare-interface', 'declare-type'],
+    ['export-interface', 'export-type'],
+    ['interface', 'type'],
+    'declare-class',
+    'export-class',
+    'class',
+    'declare-function',
+    'export-function',
+    'function',
+  ]
+  ```
+</sub>
+
+Allows you to specify a list of module member groups for sorting. Groups help organize module members into categories, prioritizing them during sorting.
+
+Each module member will be assigned a single group specified in the `groups` option (or the `unknown` group if no match is found).
+The order of items in the `groups` option determines how groups are ordered.
+
+Within a given group, members will be sorted according to the `type`, `order`, `ignoreCase`, etc. options.
+
+Individual groups can be combined together by placing them in an array. The order of groups in that array does not matter.
+All members of the groups in the array will be sorted together as if they were part of a single group.
+
+Predefined groups are characterized by a single selector and potentially multiple modifiers. You may enter modifiers in any order, but the selector must always come at the end.
+
+#### Interfaces
+
+- Selector: `interface`.
+- Modifiers: `declare`, `default`, `export`.
+- Example: `declare-interface`, `export-default-interface` or `interface`.
+
+#### Types
+
+- Selector: `type`.
+- Modifiers: `declare`, `export`.
+- Example: `declare-type`, `declare-export-type` or `interface`.
+
+#### Classes
+
+- Selector: `class`.
+- Modifiers: `declare`, `default`, `decorated`, `export`.
+- Example: `declare-class`, `export-default-decorated-class` or `class`.
+
+#### Functions
+
+- Selector: `function`.
+- Modifiers: `declare`, `default`, `async`, `export`.
+- Example: `async-function`, `export-default-function` or `function`.
```

Comment:
That's not the purpose of this PR, but could we also add `async` to `sort-classes`?

Reply:
@OlivierZal I have it planned! It will come in a separate PR.